src/stream.rs

```diff
@@ -0,0 +1,127 @@
+#[cfg(feature = "ssl")]
+use std::io::{Error, ErrorKind};
+use std::io::{Read, Result, Write};
+use std::net::{SocketAddr, TcpStream};
+
+#[cfg(feature = "ssl")]
+use openssl::ssl::{Ssl, SslMethod, SslContext, SslStream, SSL_OP_NO_COMPRESSION, SSL_OP_NO_SSLV2,
+                   SSL_OP_NO_SSLV3, SSL_VERIFY_NONE, SSL_VERIFY_PEER};
+#[cfg(feature = "ssl")]
+use openssl::x509::X509_FILETYPE_PEM;
+
+#[derive(Clone)]
+pub enum StreamConnector {
+    Tcp,
+    #[cfg(feature = "ssl")]
+    Ssl {
+        ca_file: String,
+        certificate_file: String,
+        key_file: String,
+        verify_peer: bool,
+    },
+}
+
+impl Default for StreamConnector {
+    fn default() -> Self {
+        StreamConnector::Tcp
+    }
+}
+
+impl StreamConnector {
+    #[cfg(feature = "ssl")]
+    pub fn with_ssl(ca_file: &str,
+                    certificate_file: &str,
+                    key_file: &str,
+                    verify_peer: bool)
+                    -> Self {
+        StreamConnector::Ssl {
+            ca_file: String::from(ca_file),
+            certificate_file: String::from(certificate_file),
+            key_file: String::from(key_file),
+            verify_peer: verify_peer,
+        }
+    }
+
+    pub fn connect(&self, hostname: &str, port: u16) -> Result<Stream> {
```

Comment:
gonna be that guy commenting on docs 😬. maybe document the default options for SSL as well here, to be explicit about what the user should expect?